FF124 shadowRootClonable and friends (#33325)
mdn/content: files/en-us/web/api/htmltemplateelement/shadowrootdelegatesfocus/index.md

---
title: "HTMLTemplateElement: shadowRootDelegatesFocus property"
short-title: shadowRootDelegatesFocus
slug: Web/API/HTMLTemplateElement/shadowRootDelegatesFocus
page-type: web-api-instance-property
browser-compat: api.HTMLTemplateElement.shadowRootDelegatesFocus
---

{{APIRef("Web Components")}}

The **`shadowRootDelegatesFocus`** property of the {{domxref("HTMLTemplateElement")}} interface indicates whether a [shadow root](/en-US/docs/Glossary/Shadow_tree) that that was [declaratively created](/en-US/docs/Web/HTML/Element/template#declarative_shadow_dom) will [delegate focus](/en-US/docs/Web/API/ShadowRoot/delegatesFocus).

> If the value is `true`, then selecting a non-focussable element in the [shadow tree](/en-US/docs/Glossary/Shadow_tree) (or calling `focus()` on the host element), will cause the focus to be given to the first focussable element in the tree, and the host is given any available [`:focus`](/en-US/docs/Web/CSS/:focus) styling.

Reviewer: I'm pretty sure "focussable" should be "focusable"; I did some research on it. I'll just mention this instance, but you should do a search and replace for it.

> If the value is `true`, then selecting a non-focussable element in the [shadow tree](/en-US/docs/Glossary/Shadow_tree) (or calling `focus()` on the host element), will cause the focus to be given to the first focussable element in the tree, and the host is given any available [`:focus`](/en-US/docs/Web/CSS/:focus) styling.

Author: Thanks! You're right. Will fix.

> By default the value is `false`, so if a non-focusable element in the shadow tree is selected the focus will remain with the previously focused element.

Author: FYI This was verified by testing using the example added to the https://pr33325.content.dev.mdn.mozit.cloud/en-US/docs/Web/HTML/Element/template#declarative_shadow_dom_with_delegated_focus. I set some highlight on the surrounding div if it has the focus, and this only triggers if something in the shadow root has focus (and then it always does).
Note, this wasn't clear to me in the spec though.

> By default the value is `false`.<!-- and when a non focussable element in the shadow tree is selected the shadow host (parent element) receives any [`:focus`](/en-US/docs/Web/CSS/:focus) styling such as an outline or border-->

Author: What happens to the focus the value is `false`? The focus is not delegated, but testing shows it also does not go to the parent.

> By default the value is `false`.<!-- and when a non focussable element in the shadow tree is selected the shadow host (parent element) receives any [`:focus`](/en-US/docs/Web/CSS/:focus) styling such as an outline or border-->

Author: ```suggestion
By default the value is `false`, so if a non-focusable element in the shadow tree is selected the focus will remain with the previously focused element.
```

This property reflects the value of the [`shadowrootdelegatesfocus`](/en-US/docs/Web/HTML/Element/template#shadowrootclonable) attribute of the associated [`<template>`](/en-US/docs/Web/HTML/Element/template) element.

## Value

`true` if the shadow root delegates focus; otherwise `false`.

## Specifications

{{Specifications}}

> 

Author: FYI, example of this is in https://pr33325.content.dev.mdn.mozit.cloud/en-US/docs/Web/HTML/Element/template#declarative_shadow_dom_with_delegated_focus - could link from here but wanted to see what happens with that example in review.

Reviewer: I think the example seems reasonable, and you should link it. It needs some kind of example.

Author: Done
